# HG changeset patch
# User Rik <rik@octave.org>
# Date 1551835778 28800
#      Tue Mar 05 17:29:38 2019 -0800
# Branch stable
# Node ID 4d7665fc8f6c797ad085ade5f346da346e1a0778
# Parent  55e087e232fef48f5be0375c240c3ff1457590c8
line.m: Fix creation of unwanted axes (bug #55840)

* line.m: Save/restore figure when plotting into an axes other than gca.

diff --git a/scripts/plot/draw/line.m b/scripts/plot/draw/line.m
--- a/scripts/plot/draw/line.m
+++ b/scripts/plot/draw/line.m
@@ -63,21 +63,30 @@
 
 function h = line (varargin)
 
   ## Get axis argument which may be in a 'parent' PROP/VAL pair
   [hax, varargin] = __plt_get_axis_arg__ ("line", varargin{:});
 
   if (isempty (hax))
     hax = gca ();
+    oldfig = [];
   else
     hax = hax(1);
+    oldfig = get (0, "currentfigure");
+    set (0, "currentfigure", ancestor (hax, "figure"));
   endif
 
-  htmp = __line__ (hax, varargin{:});
+  unwind_protect
+    htmp = __line__ (hax, varargin{:});
+  unwind_protect_cleanup
+    if (! isempty (oldfig))
+      set (0, "currentfigure", oldfig);
+    endif
+  end_unwind_protect
 
   if (nargout > 0)
     h = htmp;
   endif
 
 endfunction
 
 
